pkg/entity/commission.go

```diff
@@ -0,0 +1,34 @@
+package entity
+
+import (
+	"time"
+)
+
+// Commission represents a record of commission allocated on a transaction for a specific user.
+type Commission struct {
+	ID        uint
+	CreatedAt time.Time
+	UpdatedAt time.Time
+
+	UserID        int
+	TransactionID uint
+	Satoshis      uint64
+	KeyOffset     string
+	IsRedeemed    bool
+	LockingScript []byte
+}
+
+// CommissionReadSpecification defines filter criteria for querying commissions based on ID, redemption, user, or satoshi amount.
+type CommissionReadSpecification struct {
```

Comment:
Don't we want to have also filter by locking script hex and by key offset (whatever it is) and transaction id